x/auctions/keeper/hooks.go

```diff
@@ -30,21 +30,21 @@ func (k Keeper) AfterEpochEnd(ctx sdk.Context, epochIdentifier string, _ int64)
 	lastBid := k.GetHighestBid(ctx)
 
 	// Distribute the awards from the last auction
-	if isValidBid(lastBid) {
-		bidWinner, err := sdk.AccAddressFromBech32(lastBid.Sender)
-		if err != nil {
-			return
-		}
 
+	// lastBid.Sender can be "", "invalidBech32" or "validBech32".
+	bidWinner, err := sdk.AccAddressFromBech32(lastBid.Sender)
+
+	// Create a cached context that is committed only
+	// if not errors happen.
+	ctxCache, writeFn := ctx.CacheContext()
```

Comment:
the current logic is different as described here - it is always committed with the `writeFn` even if there is an error as the balance from the auctions collector is sent to the auctions module account below

Reply:
wdym? The `writeFn` is called right before returning the functions. If an error occurs before the hook exit without committing

Reply:
If `sdk.AccAddressFromBech32` returns an error we are continuing with the execution in line 62 which is setting the next round and then moves to sending the coins between the modules so `writeFn` is also called when there is an error at this moment :eyes: